extensions/wikia/MercuryApi/MercuryApiController.class.php

```diff
@@ -381,11 +387,9 @@ public function getPage() {
 						$data['article'] = $articleData;
 
 						if ( !$title->isContentPage() ) {
-							// This does two things:
-							// - gets displayTitle from parser function {{DISPLAYTITLE:displayTitle}}
-							// - removes the namespace prefix from it
-							$data['article']['displayTitle'] =
-								Title::newFromText( $data['article']['displayTitle'] )->getText();
+							// Remove the namespace prefix from display title
+							$displayTitle = Title::newFromText( $displayTitle )->getText();
```

Comment:
`Title::newFromText` throws when an object is passed. `$displayTitle` comes from `ParserOutput::getTitleText` which is supposed to return only strings.

If it throws, then something bad is happening and we will catch and log it at the end of the `getPage` method.